src/Dialogs/StripeDialog.vala

```diff
@@ -270,8 +281,17 @@ public class AppCenter.Widgets.StripeDialog : Granite.Dialog {
             amount = 10;
         });
 
+        custom_amount_controllerkey.key_released.connect ((keyval, keycode, state) => {
+            if (keyval == Gdk.Key.Return || keyval == Gdk.Key.KP_Enter) {
+                activate_default ();
+            }
+        });
+
         email_entry.changed.connect (() => {
-            email_entry.text = email_entry.text.replace (" ", "").down ();
+            if (" " in email_entry.text) {
+                email_entry.text = email_entry.text.replace (" ", "");
+            }
```

Comment:
Yes because this now only runs if there is a space so converting to lowercase is not really relevant, and we also convert what we actually submit to lowercase in `on_pay_clicked ()` so this isn't really necessary